src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/FrameConnection.cs

```diff
@@ -277,42 +302,66 @@ public void Tick(DateTimeOffset now)
                     Timeout();
                 }
             }
-            else
+        }
+
+        private void CheckForReadTimeout(long timestamp)
```

Comment:
Nit: I would rename this methods to CheckForRead*Rate*Timeout and CheckForWriteRateTimeout to make it clear that these aren't absolute timeouts.